tests/test_engine.py

```diff
@@ -68,7 +69,7 @@ def test_next_scene_none(self):
         self.assertIs(self.engine.current_scene, self.mock_scene)
 
 
-def test_scene_change():
+def test_scene_change_thrashing():
```

Comment:
Should this test be updated for the event-based way of doing things? Or maybe copied/updated?

Reply:
This shouldn't affect signal based scene changes, since it requires an action from the user instead of an implicit result of setting values. So because we won't be using `Sprite.next` to do anything (once the old code is removed) it's not a thing that should happen.